plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/rest/IcebergRestCatalogConfig.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.iceberg.catalog.rest;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.configuration.ConfigSecuritySensitive;
+
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+
+import java.net.URI;
+import java.util.Optional;
+
+public class IcebergRestCatalogConfig
+{
+    public enum Security
+    {
+        NONE,
+        OAUTH2,
+    }
+
+    private URI restUri;
+    private Security security = Security.NONE;
+    private String credential;
+    private String token;
+
+    @NotEmpty(message = "iceberg.rest.uri cannot be empty")
+    @NotNull
+    public URI getBaseUri()
+    {
+        return this.restUri;
+    }
+
+    @Config("iceberg.rest.uri")
+    @ConfigDescription("The URI to the REST server")
+    public IcebergRestCatalogConfig setBaseUri(String uri)
+    {
+        if (uri == null) {
+            this.restUri = null;
+        }
+        else {
+            this.restUri = URI.create(uri);
+        }
+        return this;
+    }
+
+    @NotNull
+    public Security getSecurity()
+    {
+        return security;
+    }
+
+    @Config("iceberg.rest.security")
+    @ConfigDescription("Authorization protocol to use when communicating with the REST catalog server")
```

Comment:
Are there other security types you see implementing? With the current options this parameter is not really necessary. If `iceberg.rest.oauth2.credential` or `iceberg.rest.oauth2.token` are set, use OAUTH, otherwise it's NONE.

If you want to leave this is I'd split `iceberg.rest.oauth2.credential` and `iceberg.rest.oauth2.token` into a separate `IcebergRestCatalogOauthConfig` and optionally bind it when security is set to OAUTH. This would prevent users from specifying the oauth parameters when the security is not enabled.

Reply:
Currently, just NONE and OAuth2 are supported and we were asked to add this flag (cc @ebyhr).  I think it does help to be explicit as to whether you want security enabled or not.

As for the token/credential as a separate config, that would just leave one config param in the REST config, so I feel like it's probably something to revisit when the next auth type is added, so we know better how to evolve it.

Reply:
Updated to include the separate security config.